### PR TITLE
fix bug: the buffer is not the raw data

### PR DESCRIPTION
**Pull request by zhuang-hao-ming**

### CODE REVIEW

File: osmpbfreader.h [hunk 1]
```diff
@@ -192,7 +192,9 @@ struct Parser {
             if(sz != blob.raw_size())
                 warn() << "  reports wrong raw_size: " << blob.raw_size() << " bytes";
 
-            memcpy(unpack_buffer, buffer, sz);
+			//memcpy(unpack_buffer, buffer, sz);
+			memcpy(unpack_buffer, blob.raw().c_str(), sz); // buffer is not the raw data
```
Comment (TeXitoi) [inline osmpbfreader.h]: The comments are useless outside this PR, please remove them. Your also need to indent with spaces only.

**Review by TeXitoi — CHANGES_REQUESTED** [review]
Great find! 👍 

I admit we never read pbf that is not zlib 😇

**Comment by zhuang-hao-ming** [comment]
Hi @TeXitoi.  As your requirement, i have added a new commit.

File: osmpbfreader.h [hunk 2]
```diff
@@ -190,9 +190,9 @@ struct Parser {
 
             // check that raw_size is set correctly
             if(sz != blob.raw_size())
-                warn() << "  reports wrong raw_size: " << blob.raw_size() << " bytes";
+                warn() << "  reports wrong raw_size: " << blob.raw_size() << " bytes";			
+                memcpy(unpack_buffer, blob.raw().c_str(), sz);
```
Comment (TeXitoi) [inline osmpbfreader.h:194]: This line is badly indented.

File: osmpbfreader.h [hunk 3]
```diff
@@ -190,9 +190,9 @@ struct Parser {
 
             // check that raw_size is set correctly
             if(sz != blob.raw_size())
-                warn() << "  reports wrong raw_size: " << blob.raw_size() << " bytes";
+                warn() << "  reports wrong raw_size: " << blob.raw_size() << " bytes";
```
Comment (TeXitoi) [inline osmpbfreader.h:193]: You added training spaces at the end of the line

**Review by TeXitoi — CHANGES_REQUESTED** [review]
Sorry, still not OK.

github should show in the diff that you only changed `buffer` to `blob.raw().c_str()`, else you added noise around.

File: osmpbfreader.h (hunk 2 above)
Comment (zhuang-hao-ming) [reply osmpbfreader.h:194]: Hi @TeXitoi .Very thanks to your review. To be truth, I am a very newbie in git. I think we can save our time by just close this PR, and you can edit the code yourself.

**Comment by TeXitoi** [comment]
I merged it by hand. Your name appear in the commit history: https://github.com/CanalTP/libosmpbfreader/commit/85d55ba2be1d8372e16a1b90e1fb2ce9c46c512a

Thanks for the fix!

**Pull request closed without merging**